src/lib/uri/uri.h

```diff
@@ -13,6 +13,52 @@
 extern "C" {
 #endif /* defined(__cplusplus) */
 
+struct uri_option_value {
+        /** Each URI option value can be represented by string. */
+        char *value;
+};
+
+struct uri_option_values {
+        /**
+         * Copies of strings, which contains one or several option values.
+         * We should work with copies of appropriate string, because lua
+         * strings are constants.
+         */
+        char **copies;
+        /** Count of `copies` array. */
+        int copies_count;
+        /** Array of URI option values. */
+        struct uri_option_value *values;
+        /** Count of URI option values. */
+        int values_count;
+};
+
+struct uri_option {
+        /**
+         * Copy of URI option string. We should work with copy of
+         * appropriate string, because lua strings are constants.
+         */
+        char *copy;
+        /** Name of URI option. */
+        const char *name;
+        /** Different values of one URI string option */
+       struct uri_option_values values;
+};
+
+struct uri_options {
```

Comment:
Do you really need this as a separate struct? Would it be better to inline it in `uri`?
`uri->options.options[i]` looks ugly IMO; better use `uri->options[i]`.